microcontroller/Core/Src/sdlog.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * sdlog.c
+ *
+ *  Created on: Apr 19, 2023
+ *      Author: Isaak
+ */
+
+#include "sdlog.h"
+
+
+
+void SD_GPIO_Init(){
+	GPIO_InitTypeDef GPIO_InitStruct;
+
+	GPIO_InitStruct.Pin = SD_SPI_CS_PIN;
+	GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+	GPIO_InitStruct.Pull = GPIO_PULLUP;
+	GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_HIGH;
+	HAL_GPIO_Init(SD_SPI_GPIO_PORT, &GPIO_InitStruct);
+}
+
+void SD_SPI_Init(){
+	GPIO_InitTypeDef GPIO_InitStruct;
+
+	// Configuration of GPIO pins for SPI communication
+
+	GPIO_InitStruct.Pin =  GPIO_PIN_8 |GPIO_PIN_9 | GPIO_PIN_10 | GPIO_PIN_11;
+	GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
+	GPIO_InitStruct.Pull = GPIO_NOPULL;
+	GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_HIGH;
+	GPIO_InitStruct.Alternate = GPIO_AF5_SPI1;
+	HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+	// Initialization of SD Card
+
+	SD_SPI_HANDLE.Instance = SPI1;
+	SD_SPI_HANDLE.Init.Mode = SPI_MODE_MASTER;
+	SD_SPI_HANDLE.Init.Direction = SPI_DIRECTION_2LINES;
+	SD_SPI_HANDLE.Init.DataSize = SPI_DATASIZE_8BIT;
+	SD_SPI_HANDLE.Init.CLKPolarity = SPI_POLARITY_LOW;
+	SD_SPI_HANDLE.Init.CLKPhase = SPI_PHASE_1EDGE;
+	SD_SPI_HANDLE.Init.NSS = SPI_NSS_SOFT;
+	SD_SPI_HANDLE.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_256;
+	SD_SPI_HANDLE.Init.FirstBit = SPI_FIRSTBIT_MSB;
+	SD_SPI_HANDLE.Init.TIMode = SPI_TIMODE_DISABLE;
+    SD_SPI_HANDLE.Init.CRCCalculation = SPI_CRCCALCULATION_DISABLE;
+	SD_SPI_HANDLE.Init.CRCPolynomial = 10;
+	HAL_SPI_Init(&SD_SPI_HANDLE);
+}
+
+void SD_Write(const char * filename, const char * data){
```

Comment:
Looks good. If you're able to, let's try to have the file type be made as a .CSV file, just so it's easy to interpret and represent data.